### PR TITLE
Fix `CLONE` cannot set `id`

### PR DESCRIPTION
**Pull request by robartsd**

### CODE REVIEW

**Comment by ArnoldSmith86** [comment]
PR-SERVER-BOT: You can play around with it here: http://212.47.248.129:3573/pr-test (or any other room on that server)

File: client/js/widgets/widget.js [hunk 1]
```diff
@@ -519,7 +519,11 @@ export class Widget extends StateManaged {
               const parent = clone.parent;
               clone.clonedFrom = w.get('id');
 
-              delete clone.id;
+              if(widgets.has(clone.id)) {
+                delete clone.id;
+                if(properties.has('id'))
```
Comment (ArnoldSmith86) [inline client/js/widgets/widget.js:524]: Does this work? It's just an object and should be `properties.id !== undefined`, right?

Comment (robartsd) [reply client/js/widgets/widget.js:524]: Yes. I was working with Set earlier and was thinking of that interface.

**Review by bjalder26 — APPROVED** [review]
This appears to work.  I even tried setting a clone to an already-used ID, and it still cloned the widget, but just with another ID, which is what I would think should happen.  So I approve through testing.

**Review by ArnoldSmith86 — APPROVED** [review]
Looks like a good solution to me.

**Pull request merged**